### PR TITLE
Onboard B2B feature related integration tests

### PR DESCRIPTION
**Pull request by sadilchamishka**
$subject

### Related Issues
- https://github.com/wso2/product-is/issues/16359

### CODE REVIEW

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/common/B2BRESTTestBase.java [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+    /**
+     * Invoke given endpointUri for POST with given body and Basic authentication, authentication credential being the
+     * authenticatingUserName and authenticatingCredential.
+     *
+     * @param endpointUri Endpoint to be invoked.
+     * @param body        Payload.
+     * @return Response of post request.
+     */
+    protected Response getResponseOfPost(String endpointUri, String body) {
+        return given().auth().preemptive().basic(authenticatingUserName, authenticatingCredential)
```
Comment (AnuradhaSK) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/common/B2BRESTTestBase.java:95]: missing new line after the method signature

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/common/B2BRESTTestBase.java [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+            errorDescriptionArgs) {
+
+        validateHttpStatusCode(response, httpStatusCode);
+        validateResponseElement(response, "code", is(errorCode));
+        validateErrorMessage(response, errorCode);
+        validateErrorDescription(response, errorCode, errorDescriptionArgs);
+    }
+
+    /**
+     * Validate http status code of the response
```
Comment (AnuradhaSK) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/common/B2BRESTTestBase.java:143]:
```suggestion
     * Validate http status code of the response.
```

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/common/B2BRESTTestBase.java [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+
+        response
+                .then()
+                .assertThat()
+                .log().ifValidationFails()
+                .statusCode(httpStatusCode);
+    }
+
+    /**
+     * Validate error description of the response, if an entry is available in RESTAPIErrors.properties
```
Comment (AnuradhaSK) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/common/B2BRESTTestBase.java:158]:
```suggestion
     * Validate error description of the response, if an entry is available in RESTAPIErrors.properties.
```

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/common/B2BRESTTestBase.java [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+     * @param response  Response.
+     * @param errorCode Error code.
+     */
+    private void validateErrorMessage(Response response, String errorCode) {
+
+        validateElementAgainstErrorProperties(response, errorCode, "message");
+    }
+
+    /**
+     * Validate elements in error response against entries in RESTAPIErrors.properties
```
Comment (AnuradhaSK) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/common/B2BRESTTestBase.java:182]:
```suggestion
     * Validate elements in error response against entries in RESTAPIErrors.properties.
```

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/organization/management/v1/OrganizationManagementBaseTest.java [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+    @BeforeClass(alwaysRun = true)
+    public void init(ITestContext context) throws Exception {
+
+        ISuite suite = context.getSuite();
+        String orgId = (String) suite.getAttribute("createdOrgId");
+        if (orgId == null) {
+            orgId = SUPER_ORGANIZATION_ID;
+        }
+        this.subOrganizationId = orgId;
+        if (this.organizationType.name().equals("SUPER_ORGANIZATION")) {
```
Comment (AnuradhaSK) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/organization/management/v1/OrganizationManagementBaseTest.java:89]: use a constant for "SUPER_ORGANIZATION"
Also, change the check as in the following format
<constants>.equals(<variable>)

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/organization/management/v1/OrganizationManagementFailureTest.java [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+
+        JSONObject organizationObject = new JSONObject();
+        String parentId;
+
+        if (organizationType.name().equals("SUPER_ORGANIZATION")) {
+            parentId = SUPER_ORGANIZATION_NAME;
+        } else {
+            parentId = subOrganizationId;
+        }
+        organizationObject.put(ORGANIZATION_NAME, "");
```
Comment (AnuradhaSK) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/organization/management/v1/OrganizationManagementFailureTest.java:78]:
```suggestion
        organizationObject.put(ORGANIZATION_NAME, StringUtils.EMPTY);
```

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/common/B2BRESTTestBase.java (hunk 1 above)
Comment (sadilchamishka) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/common/B2BRESTTestBase.java]: Fixed by - https://github.com/wso2/product-is/pull/16566/commits/177717a2f7457f7a3e486d6359b440b160ab1fe5

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/organization/management/v1/OrganizationManagementBaseTest.java (hunk 5 above)
Comment (sadilchamishka) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/organization/management/v1/OrganizationManagementBaseTest.java]: Fixed by - https://github.com/wso2/product-is/pull/16566/commits/177717a2f7457f7a3e486d6359b440b160ab1fe5

**Review by AnuradhaSK — APPROVED** [review]

**Pull request merged**